app/src/main/java/org/kinecosystem/kinit/view/SplashActivity.kt

```diff
@@ -1,84 +1,61 @@
+
     private companion object {
         private const val PLAY_SERVICES_UPDATE_REQUEST = 100
+        val TAG: String = SplashActivity::class.java.simpleName
     }
 
-    @Inject
-    lateinit var userRepository: UserRepository
-
-    private lateinit var splashViewModel: SplashViewModel
-    private val isBLackedListListener = object : OnPropertyChangedCallback() {
-        override fun onPropertyChanged(observable: Observable?, propertyId: Int) {
-            if(splashViewModel.inBlackList.get()){
-                showDialogBlackedList()
-            }
-        }
-    }
+    private var googlePlayServicesAvailable: Int? = null
+    private val splashViewModel: SplashViewModel = SplashViewModel()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         KinitApplication.coreComponent.inject(this)
         super.onCreate(savedInstanceState)
-        splashViewModel = SplashViewModel(this)
-        moveToSplashScreen()
-    }
-
-    override fun moveToSplashScreen() {
-        val binding = DataBindingUtil.setContentView<SplashLayoutBinding>(this, R.layout.splash_layout)
-        binding.model = splashViewModel
-    }
-
-    override fun moveToErrorScreen() {
-        val binding = DataBindingUtil.setContentView<OnboardingErrorCreatingWalletLayoutBinding>(this,
-                R.layout.onboarding_error_creating_wallet_layout)
-        binding.model = splashViewModel
-    }
-
-    override fun openContactSupport() {
-        SupportUtil.openEmailSupport(this, userRepository)
-    }
-
-    override fun moveToMainScreen() {
-        startActivity(MainActivity.getIntent(this))
-        finish()
-    }
-
-    override fun onPause() {
-        super.onPause()
-        splashViewModel.inBlackList.removeOnPropertyChangedCallback(isBLackedListListener)
+        googlePlayServicesAvailable = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(applicationContext)
+        setContentView(R.layout.splash_layout)
+        splashViewModel.listener = this
     }
 
     override fun onResume() {
         super.onResume()
-        checkGoogleServices()
-        splashViewModel.onResume()
-        splashViewModel.inBlackList.addOnPropertyChangedCallback(isBLackedListListener)
-        if (splashViewModel.inBlackList.get()) {
-            showDialogBlackedList()
-            splashViewModel.inBlackList.removeOnPropertyChangedCallback(isBLackedListListener)
+        if (googlePlayServicesAvailable == ConnectionResult.SUCCESS) {
+            splashViewModel.onResume()
+        } else {
+            GoogleApiAvailability
+                    .getInstance()
+                    .getErrorDialog(
+                            this,
+                            googlePlayServicesAvailable
+                                    ?: -1, SplashActivity.PLAY_SERVICES_UPDATE_REQUEST)
+                    .show()
         }
     }
 
-    private fun showDialogBlackedList() {
+    override fun moveToNextScreen() {
```

Comment:
logic for choosing which screen could be in the model.
moveToNextScreen could take which screen as a parameter